pypdf/filters.py

```diff
@@ -340,7 +341,8 @@ def decode(
         index = 0
         while True:
             if index >= len(data):
-                raise PdfStreamError("Unexpected EOD in RunLengthDecode")
+                break  # reach End Of String even if no EOD
+                # raise PdfStreamError("Unexpected EOD in ASCIIHexDecode")
```

Comment:
This seems to be a copy&paste error within the error message.

Reply:
removed